kindcat_custom.cue

```diff
@@ -10,6 +40,61 @@ package kindsys
 Custom: S={
 	_sharedKind
 
-	lineage: { name: S.machineName }
+	// appID is the unique identifier of the app which is the owner of this Custom kind.
+	// TODO: should this just be pluginID? Is there a world where an app and plugin may not share the same ID?
+	// The practical reason to have it different is to allow the appID to be shorter than the pluginID as the group
+	// is generated from the appID, and <kind name>+<group> cannot exceed 63 characters in kubernetes.
+	appID: =~"^([a-z][a-z0-9-]{0,61}[a-z0-9])$"
```

Comment:
Discussed with @IfSentient a nice compromise might be to default to use the plugin id, but in case that results in a longer group than 63 characters we could allow an override (appId or group). This will allow plugin authors to easily determine the rules, if needed.

Reply:
I've updated the PR so that now there is a `plugin` trait with an `id` field (there is a TODO on whether this should be a trait, or if `pluginID` should just be a top-level field in the kind). The `crd` trait has an optional `groupOverride` field which can be used to override using `plugin.id` as the group name.

Reply:
I've done a weird temporary compromise where `pluginID` is a top-level field now, but the `plugin` trait still exists, and `plugin.id` is set to the `pluginID` field. Not sure if we want the trait to continue to exist, or if we want to revert that and have the ID live solely in the trait. I'm somewhat unopinionated on this, because I'm not sure if we're going to ever need more plugin info in the kind--if we do, we'll want to keep the trait. If there's both the trait and pluginID fields, it seems a bit confusing to use.